docker-compose.yml

```diff
@@ -0,0 +1,104 @@
+version: '3.8'
+
+services:
+  db:
+    image: 'mysql:8.0.19'
+    restart: always
+    working_dir: /app
+    environment:
+      MYSQL_ROOT_PASSWORD: rootpassword
+      MYSQL_DATABASE: mydb
+      MYSQL_USER: user
+      MYSQL_PASSWORD: password
+    ports:
+      - "3306:3306"
+    volumes:
+      - db_data:/var/lib/mysql
+    networks:
+      - caddy
+
+  phpmyadmin:
+    image: phpmyadmin/phpmyadmin:5.0.1
+    environment:
+      PMA_HOST: db
+      PMA_PORT: 3306
+      PMA_USER: root
+      PMA_PASSWORD: rootpassword
+      MYSQL_ROOT_PASSWORD: rootpassword
+      MYSQL_PASSWORD: password
+    ports:
+      - "8080:80"
+    depends_on:
+      - db
+    networks:
+      - caddy
+
+  api:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile
+    command: >
+      sh -c "npm run prisma-up && npm run prisma-gen && npm run start"
+    ports:
+      - "3000:3000"
+    depends_on:
+      - db
+    environment:
+      DATABASE_URL: "mysql://root:rootpassword@db:3306/mydb"
+      NODE_ENV: production
+      HEX_USER_ID_OFFSET: 0
+      PORT: 3000
+      SMTP_PORT: 465
+      SMTP_HOST: ${SMTP_HOST}
+      SMTP_USERNAME: ${SMTP_USERNAME}
+      SMTP_PASSWORD: ${SMTP_PASSWORD}
+      SMTP_SENDER: ${SMTP_SENDER}
+      JWT_SECRET: ${JWT_SECRET}
+      RECAPTCHA_SECRET_KEY: ${RECAPTCHA_SECRET_KEY}
+      SIGN_IN_URL: http://localhost:5000/auth/login
+      RESET_PASSWORD_URL: http://localhost:5000/auth/password
+      CORS_ORIGIN: http://localhost:5000
+      GOOGLE_AUTH_CLIENT_ID: none
+    networks:
+      - caddy
+    labels:
+      caddy: ${PUBLIC_BACKEND_URL}
+      caddy.reverse_proxy: "{{upstreams http 3000}}"
+
+  web:
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile
+    env_file:
+      - frontend/.env
+    ports:
+      - 5000:3000
+    depends_on:
+      - api
+    networks:
+      - caddy
+    labels:
+      caddy: ${PUBLIC_FRONTEND_URL}
+      caddy.reverse_proxy: "{{upstreams http 5000}}"
+
+  caddy:
+    image: lucaslorentz/caddy-docker-proxy:ci-alpine
+    ports:
+      - 80:80
+      - 443:443
+    environment:
+      - CADDY_INGRESS_NETWORKS=caddy
+    networks:
+      - caddy
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```

Comment:
thats part of the documentation since we are adding custom domain via labelsin docker https://github.com/lucaslorentz/caddy-docker-proxy?tab=readme-ov-file#basic-usage-example-using-docker-compose